accepted/0000-workspaces.md

```diff
@@ -0,0 +1,479 @@
+- Install: In a workspaces setup users expect to be able to install all nested packages and perform the associated lifecycle scripts from the root-level, it should also be aware of sibling packages that depend on one another and link them appropriately
+- Run some npm subcommands in the context of each nested workspace
+
+## Rationale and Alternatives
+
+First and foremost there’s the alternative of leaving the problem set for userland to solve, there’s already the very popular project Lerna that provides all of these features.
+
+Also available is the alternative of supporting only the install (or bootstrap as Lerna names it) aspect of this proposal, following a less feature-rich approach but one that would still enable the basic goal of improving the user experience of managing multiple child packages but from all the feedback collected during the research phase of this RFC, this alternative is much less desirable to the community of maintainers involved.
+
+## Implementation
+
+### 1. Make npm workspace-aware
+
+We're following the lead of Yarn in supporting the `workspaces` `package.json` field which defines a list of paths, each of these paths may be a workspace itself but it also support globs.
+
+`package.json` example:
+
+```
+{
+    "name": "workspace-example",
+    "version": "1.0.0",
+    "workspaces": {
+        "packages": [
+            "packages/*"
+        ]
+    }
+}
+```
+
+`package.json` shorthand example:
+
+```
+{
+    "name": "workspace-example",
+    "version": "1.0.0",
+    "workspaces": [
+        "packages/*"
+    ]
+}
+```
+
+
+### 2. Installing dependencies across child packages
+
+Change `npm install` (arborist) behavior to make it install every nested package by default once a valid configuration is defined.
+
+Arborist should also be aware of all nested workspaces and correctly link to an internal workspace should it match the required semver version of an expected dependency anywhere in the installing tree. e.g:
+
+```
+// Given this package.json structure:
+├── package.json { "workspaces": ["dep-a", "dep-b"] }
+├── dep-a
+│   └── package.json { "dependencies": { "dep-b": "^1.0.0" } }
+└── dep-b
+    └── package.json { "version": "1.3.1" }
+
+$ npm install
+
+// Results in this symlinking structure:
+├── node_modules
+│   ├── dep-a -> ./dep-a
+│   └── dep-b -> ./dep-b
+├── dep-a
+│   └── node_modules
+│       └── dep-b -> ../../../node_modules/dep-b
+└── dep-b
+```
+
+NOTE: The final implementation of the symlinking structure might differ from the examples laid out here (which are trying to be as illustrative as possible). Arborist might end up resolving all symlinks up to the actual place instead of creating a chain of symlinks.
+
+### 3. Run commands across all child packages
+
+Make npm commands workspace-aware, so that running a command tries to run it within nested packages as long as a workspaces field is defined in `package.json`.
```

Comment:
Outside of workspace-aware `npm run`, I don't think npm needs to provide a generic `exec` command. As long as `npm ls` continues to provide parseable output (no reason to expect otherwise), I would consider that "good enough" for core.

Reply:
I commented about it in a couple of places: [here](https://github.com/npm/rfcs/pull/103#discussion_r383477860) and [here](https://github.com/npm/rfcs/pull/103#discussion_r393835251)

bottom line here is, remember to come to the [OpenRFC call tomorrow](https://github.com/npm/rfcs/issues/111) 😄